auth/store.go

```diff
@@ -324,6 +333,43 @@ func (as *authStore) UserGet(r *pb.AuthUserGetRequest) (*pb.AuthUserGetResponse,
 	return &resp, nil
 }
 
+func (as *authStore) UserRevoke(r *pb.AuthUserRevokeRequest) (*pb.AuthUserRevokeResponse, error) {
+	tx := as.be.BatchTx()
+	tx.Lock()
+	defer tx.Unlock()
+
+	_, vs := tx.UnsafeRange(authUsersBucketName, []byte(r.Name), nil, 0)
+	if len(vs) != 1 {
+		return nil, ErrUserNotFound
+	}
+
+	user := &authpb.User{}
+	err := user.Unmarshal(vs[0])
+	if err != nil {
+		return nil, err
+	}
+
+	updatedUser := &authpb.User{}
+	updatedUser.Name = user.Name
+	updatedUser.Password = user.Password
+
+	for _, role := range user.Roles {
+		if strings.Compare(role, r.Role) != 0 {
```

Comment:
maybe we want to return an error if user tries to revoke a non-existing role? (add a todo is ok)